src/ekka_rlog.erl

```diff
@@ -0,0 +1,111 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% API and management functions for asynchronous Mnesia replication
+-module(ekka_rlog).
+
+-export([ transaction/2
+        , shards/0
+        , core_nodes/0
+        , role/0
+        , node_id/0
+        , subscribe/4
+        ]).
+
+-export_type([ shard/0
+             ]).
+
+-type shard() :: atom().
+
+-type func(A) :: fun((...) -> A).
+
+-include("ekka_rlog.hrl").
+-include_lib("mnesia/src/mnesia.hrl").
+
+-include_lib("stdlib/include/ms_transform.hrl").
+
+%% @doc Perform a transaction and log changes.
+%% the logged changes are to be replicated to other nodes.
+-spec transaction(func(A), [term()]) -> mnesia:t_result(A).
+transaction(F, Args) -> do(transaction, F, Args).
+
+%% TODO: configurable
+node_id() ->
+    0.
```

Comment:
maybe we can use `node()` for this purpose.
to avoid being forced to configure an ID.

the downside is slightly bloated rlog serialisation result.